#!/usr/bin/with-contenv bash

cd "${SJVA_HOME}"

if [ "${USE_CELERY}" = "true" ]; then
    n=0
    m=5
    while true; do
        sleep $(((n**2+1)*3))
        # temporary fix for checking status of celery app
        # celery v5.0.x does not throw error code properly
        s6-setuidgid abc \
            celery -A ${CELERY_APP} inspect ping --timeout=3 2>&1 | grep -q 'No nodes replied' || break
        ((n++))
        if [ "$n" -gt "$m" ]; then
            s6-svscanctl -t /var/run/s6/services
        else
            printf "*** celery is not ready. Retrying in %2ds...(%d/%d)\n" $(((n**2+1)*3)) $n $m
        fi
    done
fi

echo "*** starting SJVA..."
s6-setuidgid abc \
    python sjva3.py \
        --repeat=${SJVA_REPEAT_COUNT} \
        --use_gevent=${USE_GEVENT} \
        --use_celery=${USE_CELERY} \
        --port=${SJVA_PORT}

EXITCODE=$?
echo "*** SJVA exited with code ${EXITCODE}"
if [ "$EXITCODE" = "0" ]; then
    echo '*** terminating all services...'
    s6-svscanctl -t /var/run/s6/services
else
    echo "*** restarting services..."
    printf $(expr $SJVA_REPEAT_COUNT + 1) > /var/run/s6/container_environment/SJVA_REPEAT_COUNT
    /etc/cont-init.d/20-config
    s6-svc -wr -r /var/run/s6/services/celery
fi 
